experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -234,6 +235,43 @@ export type ContextuallyTypedNodeData =
  */
 export type ContextuallyTypedFieldData = ContextuallyTypedNodeData | undefined;
 
+/**
+ * Information needed to interpret a subtree described {@link ContextuallyTypedNodeData} and {@link ContextuallyTypedFieldData}.
+ * @alpha
+ */
+export interface TreeDataContext {
+	/**
+	 * Schema for the document which the tree will be used in.
+	 */
+	readonly schema: SchemaDataAndPolicy;
+
+	/**
+	 * Procedural data generator for fields.
+	 * Fields which provide generators here can be omitted in the input contextually typed data.
+	 *
+	 * @remarks
+	 * TODO:
+	 * For implementers of this which are not pure (like identifier generation),
+	 * order of invocation should be made consistent and documented.
+	 * This will be important for identifier elision optimizations in tree encoding for session based identifier generation.
+	 */
+	getFieldGenerator(key: FieldKey, schema: FieldStoredSchema): undefined | FieldGenerator;
+}
+
+/**
+ * Generates field content for a MapTree on demand.
+ * @alpha
+ */
+export type FieldGenerator = () => MapTree[];
+
+/**
+ * @alpha
+ * Default getFieldGenerator function that returns undefined.
+ */
+export const defaultGetFieldGenerator = (key: FieldKey, schema: FieldStoredSchema): undefined => {
+	return;
+};
```

Comment:
When practical, use the shorter syntax for arrow functions:

```suggestion
export const defaultGetFieldGenerator = (key: FieldKey, schema: FieldStoredSchema): undefined => undefined;
```

of perhaps better add a proper named type for the getter, then its more safe to omit the parameters and return type:
```suggestion
type GetFieldGenerator = getFieldGenerator(key: FieldKey, schema: FieldStoredSchema): undefined | FieldGenerator;

export const defaultGetFieldGenerator:  GetFieldGenerator = () => undefined;
```

Reply:
added `GetFieldGenerator` type, simplified `defaultGetFieldGenerator` to use arrow functions